integration-tests/deployment/environment.go

```diff
@@ -54,6 +56,33 @@ type Environment struct {
 	Logger   logger.Logger
 }
 
+type MultiDonEnvironment struct {
```

Comment:
Would you mind explaining the concept of an Environment please? And why it is important to "group nodesets into a don and compose those dons into an Environment"?

Reply:
the original concept inspired by CCIP was dev vs testnet vs prod.

an environment is composed of nodes, contracts, and configuration that couple them

however the implementation that they have assumes that all the nodes in a given environment host all capabilities.

this doesn't work for keystone, and one why to accomodate our need is by logically partitioning the total set of nodes into subsets that host given sets of capabilities.

so, i leveraged the existing idea of an Env and extended it to support nodesets that host different capabilities. i modeled each of these nodesets as an Environment, although it a bit of an abuse of the original idea. but that really isn't exposed. rather, the thing that is exposed is a MultiDonEnvironment, which, again aims to group nodesets into dons that host different capabilities.